File: Java/problem1394-1.java
class Solution {
    public int findLucky(int[] arr) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int a : arr) {
            if (map.containsKey(a)) {
                map.put(a, map.get(a) + 1);
            }
            else {
                map.put(a, 1);
            }
        }
        int result = -1;
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getKey() == entry.getValue()) {
                if (entry.getKey() > result) {
                    result = entry.getKey();
                }
            }
        }
        return result;
    }
}

File: python/problem1394-1.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        num2count = {}
        for a in arr:
            if a in num2count:
                num2count[a] += 1
            else:
                num2count[a] = 1
        result = -1
        for num, count in num2count.items():
            if num == count:
                result = max(result, num)
        return result
